File: uri/ad-hock/spiral/1935.cpp
#include <iostream>

using namespace std;

int main() {
    long long N, B;
    cin >> N >> B;
    
    long long layer = 0;
    long long total = 0;
    
    while (true) {
        long long side = N - 2 * layer;
        if (side <= 0) break;
        
        long long perimeter;
        if (side == 1) {
            perimeter = 1;
        } else {
            perimeter = 4 * side - 4;
        }
        
        if (total + perimeter >= B) {
            break;
        }
        
        total += perimeter;
        layer++;
    }
    
    long long offset = B - total - 1;
    long long side = N - 2 * layer;
    long long row = 1 + layer;
    long long col = 1 + layer;
    
    if (side == 1) {
        cout << row << " " << col << endl;
        return 0;
    }
    
    if (offset < side - 1) {
        row = 1 + layer;
        col = 1 + layer + offset;
    } else if (offset < 2 * (side - 1)) {
        row = 1 + layer + (offset - (side - 1));
        col = N - layer;
    } else if (offset < 3 * (side - 1)) {
        row = N - layer;
        col = N - layer - (offset - 2 * (side - 1));
    } else {
        row = N - layer - (offset - 3 * (side - 1));
        col = 1 + layer;
    }
    
    cout << row << " " << col << endl;
    
    return 0;
}

File: uri/ad-hock/spiral/1935.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let nums: Vec<i64> = input
        .trim()
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();
    
    let n = nums[0];
    let b = nums[1];
    
    let mut layer = 0;
    let mut total = 0;
    
    loop {
        let side = n - 2 * layer;
        if side <= 0 {
            break;
        }
        
        let perimeter = if side == 1 {
            1
        } else {
            4 * side - 4
        };
        
        if total + perimeter >= b {
            break;
        }
        
        total += perimeter;
        layer += 1;
    }
    
    let offset = b - total - 1;
    let side = n - 2 * layer;
    let mut row = 1 + layer;
    let mut col = 1 + layer;
    
    if side == 1 {
        println!("{} {}", row, col);
        return;
    }
    
    if offset < side - 1 {
        row = 1 + layer;
        col = 1 + layer + offset;
    } else if offset < 2 * (side - 1) {
        row = 1 + layer + (offset - (side - 1));
        col = n - layer;
    } else if offset < 3 * (side - 1) {
        row = n - layer;
        col = n - layer - (offset - 2 * (side - 1));
    } else {
        row = n - layer - (offset - 3 * (side - 1));
        col = 1 + layer;
    }
    
    println!("{} {}", row, col);
}

